store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java

```diff
@@ -53,41 +53,60 @@
     RealInternalStore(Fetcher<Raw, Key> fetcher,
                       Persister<Raw, Key> persister,
                       KeyParser<Key, Raw, Parsed> parser,
-                      Cache<Key, Observable<Parsed>> memCache,
                       StalePolicy stalePolicy) {
-        init(fetcher, persister, parser, memCache, stalePolicy);
+        init(fetcher, persister, parser, stalePolicy);
     }
 
 
     RealInternalStore(Fetcher<Raw, Key> fetcher,
                       Persister<Raw, Key> persister,
                       KeyParser<Key, Raw, Parsed> parser,
+                      long expireAfter,
+                      TimeUnit expireAfterTimeUnit,
                       StalePolicy stalePolicy) {
-        memCache = CacheBuilder.newBuilder()
-                .maximumSize(getCacheSize())
-                .expireAfterAccess(getCacheTTL(), TimeUnit.SECONDS)
+        init(fetcher, persister, parser, expireAfter, expireAfterTimeUnit, stalePolicy);
+    }
+
+    private void init(Fetcher<Raw, Key> fetcher,
+                      Persister<Raw, Key> persister,
+                      KeyParser<Key, Raw, Parsed> parser,
+                      long expireAfter,
+                      TimeUnit expireAfterTimeUnit,
+                      StalePolicy stalePolicy) {
+        this.fetcher = fetcher;
+        this.persister = persister;
+        this.parser = parser;
+        this.memCache = CacheBuilder
+                .newBuilder()
+                .expireAfterWrite(expireAfter, expireAfterTimeUnit)
+                .build();
+        this.stalePolicy = stalePolicy;
+        this.inFlightRequests = CacheBuilder.newBuilder()
+                .expireAfterWrite(expireAfter, expireAfterTimeUnit)
```

Comment:
in flight request should not necessarily be same as memory, its more that it should be the max(1 minute, memory limit)